### ISSUE TITLE
Unhandled HTTP 401 from requests.get on tracks from artists that have disabled Soundcloud API access

### ISSUE THREAD

**Issue by bsans** [Miserlou/SoundScrape#71]
At first I thought this was an error on all API attempts, maybe related to recent changes to Soundcloud from them adding the "Soundcloud Go" product.

I was trying to do:

```
~/projects/SoundScrape/soundscrape$ ./soundscrape.py -f https://soundcloud.com/zhu/in-the-morning
Expecting value: line 1 column 1 (char 0)
```

which is coming from the attempted JSON decoding of an HTTP 401:

```
(Pdb) requests.get(streams_url).json()
*** JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

the 401 shows up from:

```
(Pdb) requests.get(streams_url)
<Response [401]>
```

However, a little reading at Soundcloud forums turns up this:
[https://www.soundcloudcommunity.com/soundcloud/topics/soundcloud-go-api-access](https://www.soundcloudcommunity.com/soundcloud/topics/soundcloud-go-api-access)
which says:
>Thanks for writing on our help forum. I've done some research in that area and it is actually up to the uploading artist / distributor whether their content is fully available via the API. 

I confirmed that another track randomly selected from my stream is perfectly downloadable, just as it's always been since I've been (happily) using Soundscrape.

So basically Zhu has locked down his track access via the API.  😢 

As far as the issue here with the code, maybe we could handle the 401 and provide a better error message than just letting it fall through to a cryptic `Expecting value: line 1 column 1 (char 0)`?

**Comment by Miserlou** [comment]
Grr. More SC Go relate shit. Thanks for reporting.

Basically, the whole application needs to be refactored to no-longer use to the SC API, and to use scraping like we do with all our other supported services. It's a big pain in the ass. Sorry.

**Comment by bsans** [comment]
@Miserlou no worries, happy to help.  While skimming the code, I did notice there was a separate approach (scraping), but didn't read into it.  I could try to help, but haven't scraped in this way via Python

**Comment by Miserlou** [comment]
Fixed, takk.
